In [1]:
import pandas as pd
import os


In [2]:
data = pd.read_csv("../data/BioLiP_results.txt", sep="\t", header=None)

/var/folders/pn/9zlz5sfd2gbgctqwzvdl9zg40000gn/T/ipykernel_84687/1902281247.py:1: DtypeWarning: Columns (13,14,16,19) have mixed types. Specify dtype option on import or set low_memory=False.
  data = pd.read_csv("../data/BioLiP_results.txt", sep="\t", header=None)


In [4]:
data[0]

0         101m
1         102m
2         103m
3         104m
4         105m
          ... 
496955    9xia
496956    9xim
496957    9xim
496958    9xim
496959    9xim
Name: 0, Length: 496960, dtype: object

In [7]:
data[7][0]

'F43 R45 V68 S92 H93 H97 I99 Y103'

In [8]:
data

,0,1,2,3,4,5,6,7,8,9,...,11,12,13,14,15,16,17,18,19,20
0,101m,A,2.07,BS01,HEM,A,1,F43 R45 V68 S92 H93 H97 I99 Y103,F44 R46 V69 S93 H94 H98 I100 Y104,NaN,...,"1.11.1.-,1.7.-.-","0004601,0005344,0005737,0015671,0016491,001652...",NaN,NaN,NaN,NaN,P02185,NaN,155,MVLSEGEWQLVLHVWAKVEADVAGHGQDILIRLFKSHPETLEKFDR...
1,102m,A,1.84,BS01,HEM,A,1,F43 R45 T67 L89 S92 H93 H97 I99 Y103,F44 R46 T68 L90 S93 H94 H98 I100 Y104,NaN,...,"1.11.1.-,1.7.-.-","0004601,0005344,0005737,0015671,0016491,001652...",NaN,NaN,NaN,NaN,P02185,NaN,155,MVLSEGEWQLVLHVWAKVEADVAGHGQDILIRLFKSHPETLEKFDR...
2,103m,A,2.07,BS01,HEM,A,1,F43 R45 S92 H93 H97 I99 Y103,F44 R46 S93 H94 H98 I100 Y104,NaN,...,"1.11.1.-,1.7.-.-","0004601,0005344,0005737,0015671,0016491,001652...",NaN,NaN,NaN,NaN,P02185,NaN,155,MVLSEGEWQLVLHVWAKVEADVAGHGQDILIRLFKSHPETLEKFDR...
3,104m,A,1.71,BS01,HEM,A,1,F43 R45 V68 S92 H93 H97 I99 Y103 F138,F43 R45 V68 S92 H93 H97 I99 Y103 F138,NaN,...,"1.11.1.-,1.7.-.-","0004601,0005344,0005737,0015671,0016491,001652...",NaN,NaN,NaN,NaN,P02185,NaN,155,VLSEGEWQLVLHVWAKVEADVAGHGQDILIRLFKSHPETLEKFDRF...
4,105m,A,2.02,BS01,HEM,A,1,F43 R45 H64 V68 L89 H93 H97 I99,F43 R45 H64 V68 L89 H93 H97 I99,NaN,...,"1.11.1.-,1.7.-.-","0004601,0005344,0005737,0015671,0016491,001652...",NaN,NaN,NaN,NaN,P02185,NaN,155,VLSEGEWQLVLHVWAKVEADVAGHGQDILIRLFKSHPETLEKFDRF...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
496955,9xia,A,1.90,BS01,DFR,A,1,W16 H54 M88 T90 V135 W137 E181 D245 D287,W16 H54 M88 T90 V135 W137 E181 D245 D287,H54 D57 M88 E181 K183 E217 H220 D245 D255 D257...,...,5.3.1.5,"0000287,0005737,0005975,0009045,0016853,004273...",NaN,NaN,NaN,NaN,P24300,2734296.0,389,MNYQPTPEDRFTFGLWTVGWQGRDPFGDATRRALDPVESVQRLAEL...
496956,9xim,A,2.40,BS01,XLS,A,1,H54 W137 E181 K183 E217 H220 D292,H52 W135 E179 K181 E215 H218 D290,H54 D57 M88 E181 K183 E217 H220 D245 D255 D257...,...,5.3.1.5,"0000287,0005737,0005975,0009045,0016853,004273...",NaN,NaN,NaN,NaN,P12851,1610791.0,397,VQATREDKFSFGLWTVGWQARDAFGDATRTALDPVEAVHKLAEIGA...
496957,9xim,B,2.40,BS01,XLS,B,1,H54 W137 E181 K183 H220 D292,H52 W135 E179 K181 H218 D290,H54 D57 M88 E181 K183 E217 H220 D245 D255 D257...,...,5.3.1.5,"0000287,0005737,0005975,0009045,0016853,004273...",NaN,NaN,NaN,NaN,P12851,1610791.0,397,VQATREDKFSFGLWTVGWQARDAFGDATRTALDPVEAVHKLAEIGA...
496958,9xim,C,2.40,BS01,XLS,C,1,H54 W137 E181 K183 H220 D292,H51 W134 E178 K180 H217 D289,H54 D57 M88 E181 K183 E217 H220 D245 D255 D257...,...,5.3.1.5,"0000287,0005737,0005975,0009045,0016853,004273...",NaN,NaN,NaN,NaN,P12851,1610791.0,397,QATREDKFSFGLWTVGWQARDAFGDATRTALDPVEAVHKLAEIGAY...



Each line in the annotation file BioLiP_*.txt annotates for each ligand-protein interaction site in BioLiP. 
The columns are separated by the tab key, i.e., "\t", so that you can easily import the data into MySQL database with command like this:

load data local infile 'BioLiP.dat' into TABLE biolip FIELDS TERMINATED BY '\t';



Example entry: 
966c    A       1.90    BS06    RS2     A       1       N180 L181 A182 V215 H218 E219 H222 H228 L235 Y237 P238 S239 Y240 T241   N73 L74 A75 V108 H111 E112 H115 H121 L128 Y130 P131 S132 Y133 T134      M236 E219;E219  M129 E112;E112  3.4.24.-        0004222,0006508,0008237,0008270,0031012         ki=23nM (RS2)   Ki=23nM (RS2)           P03956  10074939        RWEQTHLTYRIENYTPDLPRADVDHAIEKAFQLWSNVTPLTFTKVSEGQADIMISFVRGDHRDNSPFDGPGGNLAHAFQPGPGIGGDAHFDEDERWTNNFREYNLHRVAAHELGHSLGLSHSTDIGALMYPSYTFSGDVQLAQDDIDGIQAIYGRSQ

The columns are (from left to right):   

01	PDB ID.  `
02	Receptor chain.   
03	Resolution. "-1.00" stands for lack of resolution information, e.g. for NMR.   
04	Binding site number code.   
05	Ligand ID in the Chemical Component Dictionary (CCD) used by the PDB database.  
06	Ligand chain.  
07	Ligand serial number.   
08      Binding site residues (with PDB residue numbering).   
09      Binding site residues (with residue re-numbered starting from 1).   
10	Catalytic site residues (different sites are separated by ';') (with PDB residue numbering).   
11      Catalytic site residues (different sites are separated by ';') (with residue re-numbered starting from 1).   
12	EC number.   
13	GO terms.   
14	Binding affinity by manual survey of the original literature. The information in '()' is the PubMed ID.   
15	Binding affinity provided by the Binding MOAD database. The information in '()' is the ligand information in Binding MOAD.   
16	Binding affinity provided by the PDBbind-CN database. The information in '()' is the ligand information in PDBbind-CN.   
17	Binding affinity provided by the BindingDB database.   
18	UniProt ID.   
19	PubMed ID.   
20	Residue sequence number of the ligand (field _atom_site.auth_seq_id in PDBx/mmCIF format).   
21	Receptor sequence.   

The ligand-protein complex structure for each line entry be obtained by using the two PDB files:
1. the receptor structure file under the "receptor" folder (name is formed with columns 01,02: i.e., 0102.pdb)
2. the corresponding ligand structure file under the "ligand" folder (name is formed with columns 01,05,06,07: i.e., 01_05_06_07.pdb)

For example, the complex structure for the above example can be obtained using the two files:
1. 966cA.pdb under the "receptor" folder
2. 966c_RS2_A_1.pdb under the "ligand" folder


Please kindly cite the following article if you use BioLiP:

Jianyi Yang, Ambrish Roy, Yang Zhang, BioLiP: a semi-manually curated database for biologically relevant ligand-protein interactions, Nucleic Acids Research, 41:D1096-D1103, 2013. 